system-tests/cli/cli.test.js

```diff
@@ -105,4 +106,42 @@ describe('CLI', () => {
 			expect.stringContaining('Unexpected empty source'),
 		);
 	});
+
+	it('helps windows users with erroneous single quotes in their package.json', async () => {
+		const spy = jest.spyOn(os, 'platform').mockImplementation(() => 'win32');
+
+		return Promise.resolve(cli([`'${path.join(__dirname, 'stylesheet.css')}'`])).then(() => {
+			expect(console.log).toHaveBeenNthCalledWith(
+				1,
+				expect.stringContaining(`stylesheet.css'" were found.
+
+Please use escaped double quotes for a file pattern in your package.json, e.g.:
+
+	stylelint \\"file.css\\"
+`),
+			);
+
+			spy.mockRestore();
+		});
```

Comment:
I think we can avoid promise here and use async-await instead:

```js
await cli([`'${path.join(__dirname, 'stylesheet.css')}'`]);

expect(console.log).toHaveBeenNthCalledWith(
	1,
	expect.stringContaining(`stylesheet.css'" were found.
Please use escaped double quotes for a file pattern in your package.json, e.g.:
	stylelint \\"file.css\\"
`),
);

spy.mockRestore();
```

Same could be applied to the next test.

This way we will have less nesting, and consistent tests implementation in this file.

Reply:
Makes total sense. Will work on it ASAP!

Reply:
I've had a few problems with this because of how we're detecting the run (using the `npm_lifecycle_event` which is perhaps what @evilebottnawi was concerned about. I think I might need to completely re-work the mechanism here. I'll try and take another stab at it as time allows.

Reply:
> I'll try and take another stab at it as time allows.

Sounds good to me.